packages/mobile/src/components/withdraw-usdc-drawer/WithdrawUSDCDrawer.tsx

```diff
@@ -0,0 +1,215 @@
+import type { BNUSDC } from '@audius/common/models'
+import { Status } from '@audius/common/models'
+import { FeatureFlags } from '@audius/common/services'
+import {
+  useWithdrawUSDCModal,
+  WithdrawMethod,
+  withdrawUSDCActions,
+  WithdrawUSDCModalPages,
+  withdrawUSDCSelectors,
+  buyUSDCSelectors,
+  isValidSolAddress
+} from '@audius/common/store'
+import { formatUSDCWeiToFloorCentsNumber } from '@audius/common/utils'
+import BN from 'bn.js'
+import type { FormikProps } from 'formik'
+import { Formik, useFormikContext } from 'formik'
+import { useDispatch, useSelector } from 'react-redux'
+import { z } from 'zod'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
+
+import { Divider, Flex, Text } from '@audius/harmony-native'
+import Drawer from 'app/components/drawer'
+
+import { CoinflowConfirmTransfer } from './components/CoinflowConfirmTransfer'
+import { CryptoConfirmTransfer } from './components/CryptoConfirmTransfer'
+import { EnterTransferDetails } from './components/EnterTransferDetails'
+import { PrepareTransfer } from './components/PrepareTransfer'
+import { TransferInProgress } from './components/TransferInProgress'
+import { TransferSuccessful } from './components/TransferSuccessful'
+import {
+  AMOUNT,
+  METHOD,
+  ADDRESS,
+  DESTINATION,
+  CONFIRM,
+  type WithdrawFormValues
+} from './types'
+
+const MINIMUM_MANUAL_TRANSFER_AMOUNT_CENTS = 1
+
+const { beginWithdrawUSDC } = withdrawUSDCActions
+const { getWithdrawStatus } = withdrawUSDCSelectors
+const { getRecoveryStatus } = buyUSDCSelectors
+
+// Pages where the drawer should not be closable
+const DISABLE_DRAWER_CLOSE_PAGES = new Set([
+  WithdrawUSDCModalPages.PREPARE_TRANSFER,
+  WithdrawUSDCModalPages.TRANSFER_IN_PROGRESS,
+  WithdrawUSDCModalPages.COINFLOW_TRANSFER
+])
+
+// Note: TRANSFER_SUCCESSFUL is intentionally not included as users should be able to close after success
+
+const WithdrawUSDCFormSchema = (userBalanceCents: number) => {
+  const amount = z
+    .number()
+    .min(
+      MINIMUM_MANUAL_TRANSFER_AMOUNT_CENTS,
+      walletMessages.errors.amountTooLow
+    )
+    .max(userBalanceCents, walletMessages.errors.insufficientBalance)
+
+  return z.discriminatedUnion(METHOD, [
+    z.object({
+      [METHOD]: z.literal(WithdrawMethod.COINFLOW),
+      [AMOUNT]: userBalanceCents !== 0 ? amount : z.number(),
+      [ADDRESS]: z.string().optional(),
+      [DESTINATION]: z.string().optional(),
+      confirm: z.boolean().optional()
+    }),
+    z.object({
+      [METHOD]: z.literal(WithdrawMethod.MANUAL_TRANSFER),
+      [AMOUNT]: userBalanceCents !== 0 ? amount : z.number(),
+      [ADDRESS]: z
+        .string()
+        .min(1, walletMessages.destinationRequired)
+        .refine(isValidSolAddress, walletMessages.errors.invalidAddress),
+      [DESTINATION]: z.string().optional(),
+      [CONFIRM]: z.boolean().optional()
+    })
+  ])
+}
+
+const WithdrawForm = ({ onClose }: { onClose: () => void }) => {
```

Comment:
this is chill as is, but would we want it to have stack navigation between the form states? check out TipArtistModalScreen for an example. The benefit for this is animation forward and backward, but this also looks very clean so lmk

Reply:
let's see if design asks for it, imo nbd